comp/api/api/apiimpl/api_test.go

```diff
@@ -0,0 +1,125 @@
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/flare/flareimpl"
+	"github.com/DataDog/datadog-agent/comp/core/hostname/hostnameinterface"
+	"github.com/DataDog/datadog-agent/comp/core/log/logimpl"
+	"github.com/DataDog/datadog-agent/comp/core/secrets"
+	"github.com/DataDog/datadog-agent/comp/core/secrets/secretsimpl"
+	"github.com/DataDog/datadog-agent/comp/core/status/statusimpl"
+	"github.com/DataDog/datadog-agent/comp/core/tagger"
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	"github.com/DataDog/datadog-agent/comp/dogstatsd/replay"
+	dogstatsdServer "github.com/DataDog/datadog-agent/comp/dogstatsd/server"
+	dogstatsddebug "github.com/DataDog/datadog-agent/comp/dogstatsd/serverDebug/serverdebugimpl"
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatformreceiver/eventplatformreceiverimpl"
+	logsAgent "github.com/DataDog/datadog-agent/comp/logs/agent"
+	"github.com/DataDog/datadog-agent/comp/metadata/host/hostimpl"
+	"github.com/DataDog/datadog-agent/comp/metadata/inventoryagent/inventoryagentimpl"
+	"github.com/DataDog/datadog-agent/comp/metadata/inventorychecks/inventorychecksimpl"
+	"github.com/DataDog/datadog-agent/comp/metadata/inventoryhost/inventoryhostimpl"
+	"github.com/DataDog/datadog-agent/comp/metadata/packagesigning/packagesigningimpl"
+	"github.com/DataDog/datadog-agent/comp/remote-config/rcservice"
+	"github.com/DataDog/datadog-agent/comp/remote-config/rcserviceha"
+
+	// package dependencies
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+
+	// third-party dependencies
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/fx"
+)
+
+func getProvides(t *testing.T) api.Component {
+	return newAPIServer(
+		fxutil.Test[dependencies](
+			t,
+			hostnameinterface.MockModule(),
+			logimpl.MockModule(),
+			config.MockModule(),
+			flareimpl.MockModule(),
+			dogstatsdServer.MockModule(),
+			replay.MockModule(),
+			dogstatsddebug.MockModule(),
+			hostimpl.MockModule(),
+			inventoryagentimpl.MockModule(),
+			demultiplexerimpl.MockModule(),
+			inventoryhostimpl.MockModule(),
+			secretsimpl.MockModule(),
+			fx.Provide(func(secretMock secrets.Mock) secrets.Component {
+				component := secretMock.(secrets.Component)
+				return component
+			}),
+			inventorychecksimpl.MockModule(),
+			packagesigningimpl.MockModule(),
+			statusimpl.MockModule(),
+			eventplatformreceiverimpl.MockModule(),
+			fx.Provide(func() optional.Option[rcservice.Component] {
+				return optional.NewNoneOption[rcservice.Component]()
+			}),
+			fx.Provide(func() optional.Option[rcserviceha.Component] {
+				return optional.NewNoneOption[rcserviceha.Component]()
+			}),
+			fetchonlyimpl.MockModule(),
+		),
+	)
+}
+
+func getTestAPIServer(t *testing.T) *apiServer {
+	p := getProvides(t)
+	return p.(*apiServer)
```

Comment:
Do we need to return the private struct rather the component? Are we not testing the component functions?

Reply:
this is [how it is implemented](https://github.com/DataDog/datadog-agent/blob/main/comp/api/api/apiimpl/api.go#L87) currently, I wanted to get tests around it before making changes